contracts/VaultV3.vy

```diff
@@ -720,7 +720,6 @@ def _update_debt(strategy: address, target_debt: uint256) -> uint256:
         # if insufficient funds to deposit, transfer only what is free
         if assets_to_deposit > available_idle:
             assets_to_deposit = available_idle
-            new_debt = current_debt + assets_to_deposit
 
         if assets_to_deposit > 0:
```

Comment:
yes thats a good idea, could cause major accounting issues if it takes less than the vault tried. It should be accounted for it the strategy.maxDeposit() call but probably good to add a check just in case. 

Something like on L693 with a pre and post balance should do that right?